dot/types/babe.go

```diff
@@ -114,3 +114,27 @@ func GetSlotFromHeader(header *Header) (uint64, error) {
 
 	return slotNumber, nil
 }
+
+// IsPrimary returns true if the block was authored in a primary slot, false otherwise.
+func IsPrimary(header *Header) (bool, error) {
+	if len(header.Digest.Types) == 0 {
```

Comment:
Should there be a nil check for header here as well?

Reply:
yeah I can add it, hopefully doesn't happen though lol